src/Login.jsx

```diff
@@ -8,28 +11,15 @@ const Login = ({ onLogin }) => {
 
   const handleLogin = async () => {
     try {
-      const response = await fetch('https://reqres.in/api/login', {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ email, password }),
-      });
-
-      const data = await response.json();
+      const data = await postLogin({ email, password });
```

Comment:
one of the most relevant things to master is knowing when to split the code and when no to.

So, we have this component , the `<Login />` form. which renders the thing and fetches the API , sending the username and the password to the API, the API would return if the username+pass is ok or not. Depending on the response this component will call the father (<App />) or render some red msg.

I'd say , you can extract the API-connection-thing to a different function. It's common to have , next to components, a services folder, so i created `services/api.js` , here we could - eventually - have a complete menu (API) of methods for fetching or sending data to the API (the service)